src/connectors/azure_log.py

```diff
@@ -0,0 +1,374 @@
+        ('PROPERTIES_ACTIVITY_DISPLAY_NAME', 'VARCHAR'),
+        ('PROPERTIES_ADDITIONAL_DETAILS', 'VARIANT'),
+        ('PROPERTIES_CATEGORY', 'VARCHAR'),
+        ('PROPERTIES_ID', 'VARCHAR'),
+        ('PROPERTIES_INITIATED_BY', 'VARIANT'),
+        ('PROPERTIES_LOGGED_BY_SERVICE', 'VARCHAR'),
+        ('PROPERTIES_OPERATION_TYPE', 'VARCHAR'),
+        ('PROPERTIES_RESULT', 'VARCHAR'),
+        ('PROPERTIES_RESULT_REASON', 'VARCHAR'),
+        ('PROPERTIES_TARGET_RESOURCES', 'VARIANT'),
+        ('RESOURCE_ID', 'VARCHAR'),
+        ('RESULT_SIGNATURE', 'VARCHAR'),
+        ('TENANT_ID', 'VARCHAR'),
+        ('EVENT_TIME', 'TIMESTAMP_LTZ(9)'),
+        ('LOADED_ON', 'TIMESTAMP_LTZ(9)')
+    ],
+    'signin': [
+        ('RAW', 'VARIANT'),
+        ('HASH_RAW', 'NUMBER'),
+        ('LEVEL', 'NUMBER'),
+        ('CALLER_IP_ADDRESS', 'VARCHAR'),
+        ('CATEGORY', 'VARCHAR'),
+        ('CORRELATION_ID', 'VARCHAR'),
+        ('DURATION_MS', 'NUMBER'),
+        ('IDENTITY', 'VARCHAR'),
+        ('LOCATION', 'VARCHAR'),
+        ('OPERATION_NAME', 'VARCHAR'),
+        ('OPERATION_VERSION', 'VARCHAR'),
+        ('PROPERTIES', 'VARIANT'),
+        ('PROPERTIES_APP_DISPLAY_NAME', 'VARCHAR'),
+        ('PROPERTIES_APP_ID', 'VARCHAR'),
+        ('PROPERTIES_APPLIED_CONDITIONAL_ACESS_POLICIES', 'VARIANT'),
+        ('PROPERTIES_AUTHENTICATION_METHODS_USED', 'VARIANT'),
+        ('PROPERTIES_AUTHENTICATION_PROCESSING_DETAILS', 'VARIANT'),
+        ('PROPERTIES_CLIENT_APP_USED', 'VARCHAR'),
+        ('PROPERTIES_CONDITIONAL_ACCESS_STATUS', 'VARCHAR'),
+        ('PROPERTIES_CREATED_DATE_TIME', 'TIMESTAMP_LTZ'),
+        ('PROPERTIES_DEVICE_DETAIL', 'VARIANT'),
+        ('PROPERTIES_ID', 'VARCHAR'),
+        ('PROPERTIES_IP_ADDRESS', 'VARCHAR'),
+        ('PROPERTIES_IS_INTERACTIVE', 'BOOLEAN'),
+        ('PROPERTIES_LOCATION', 'VARIANT'),
+        ('PROPERTIES_MFA_DETAIL', 'VARIANT'),
+        ('PROPERTIES_NETWORK_LOCATION', 'VARIANT'),
+        ('PROPERTIES_PROCESSING_TIME_IN_MILLISECONDS', 'NUMBER'),
+        ('PROPERTIES_RESOURCE_DISPLAY_NAME', 'VARCHAR'),
+        ('PROPERTIES_RESOURCE_ID', 'VARCHAR'),
+        ('PROPERTIES_RISK_DETAIL', 'VARCHAR'),
+        ('PROPERTIES_RISK_EVENT_TYPES', 'VARIANT'),
+        ('PROPERTIES_RISK_LEVEL_AGGREGATED', 'VARCHAR'),
+        ('PROPERTIES_RISK_LEVEL_DURING_SIGNIN', 'VARCHAR'),
+        ('PROPERTIES_RISK_STATE', 'VARCHAR'),
+        ('PROPERTIES_STATUS', 'VARIANT'),
+        ('PROPERTIES_TOKEN_ISSUER_TYPE', 'VARCHAR'),
+        ('PROPERTIES_USER_DISPLAY_NAME', 'VARCHAR'),
+        ('PROPERTIES_USER_ID', 'VARCHAR'),
+        ('PROPERTIES_USER_PRINCIPAL_NAME', 'VARCHAR'),
+        ('RESOURCE_ID', 'VARCHAR'),
+        ('RESULT_DESCRIPTION', 'VARCHAR'),
+        ('RESULT_SIGNATURE', 'VARCHAR'),
+        ('RESULT_TYPE', 'VARCHAR'),
+        ('TENANT_ID', 'VARCHAR'),
+        ('EVENT_TIME', 'TIMESTAMP_LTZ'),
+        ('LOADED_ON', 'TIMESTAMP_LTZ')
+    ]
+}
+
+
+def connect(name, options):
+    try:
```

Comment:
exceptions are only for exceptional behavior, and this is be definition expected, since you're catching it right away, so the right construct to use here is `if 'account_name` in options`. that said, these should already be declared in `CONNECTION_OPTIONS` as required -- can we just iterate over that, outside the connector?

Reply:
added these to base branch